content/en/docs/tasks/access-kubernetes-api/custom-resource-definition-versioning.md

```diff
@@ -0,0 +1,154 @@
+---
+title: Extend the Kubernetes API with CustomResourceDefinitions
+reviewers:
+- mbohlool
+- sttts
+- liggitt
+content_template: templates/task
+---
+
+{{% capture overview %}}
+This page explains how
+[CustomResourceDefinition](/docs/reference/generated/kubernetes-api/{{< param "version" >}}/#customresourcedefinition-v1beta1-apiextensions) versioning works and how to upgrade from one version to another. 
+{{% /capture %}}
+
+{{% capture prerequisites %}}
+
+{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}
+
+* Make sure your Kubernetes cluster has a master version of 1.11.0 or higher.
+
+* Read about [custom resources](/docs/concepts/api-extension/custom-resources/).
+
+{{% /capture %}}
+
+{{% capture body %}}
+Kubernetes standard API types has versioning to let developers extend them and roll out new features. To extend kubernetes with
+Custom Resources, you should be able to do the same by supporting multiple versions of Custom Resources. CustomResourceDefinition API supports a `versions` field that can be used to list all versions for that object. Note that previous `version` field is deprecated and optional but if it is not empty, it must be the first item in the `versions` field.
+
+For example, a CustomResourceDefinition with two versions would look like this:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  # name must match the spec fields below, and be in the form: <plural>.<group>
+  name: crontabs.stable.example.com
+spec:
+  # group name to use for REST API: /apis/<group>/<version>
+  group: stable.example.com
+  # list of versions supported by this CustomResourceDefinition
+  versions:
+  - Name: v1
+    # Each version can be enabled/disabled by Served flag.
+    Served: true
+    # One and only one version must be marked as storage version.
+    storage: true
+  - Name: v2
```

Comment:
would use v1beta1 instead. v1 and v2 are probably not the prime examples for versions without conversions.